core/trino-main/src/main/java/io/trino/execution/SummaryStatistic.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS)
+public interface SummaryStatistic
+{
+    boolean isEmpty();
+
+    Type getType();
+
+    enum Type {
```

Comment:
Do we need a type? Can we use `class` instead?

Reply:
This felt a bit cleaner than using `class`. It also allows using an EnumMap in `VersionedSummaryStatsCollector` which is more efficient than normal map.